vllm/v1/worker/gpu_model_runner.py

```diff
@@ -2570,25 +2626,28 @@ def get_kv_cache_spec(self) -> dict[str, KVCacheSpec]:
                 continue
 
             # TODO: Support other attention modules, e.g., cross-attention
-            if attn_module.attn_type == AttentionType.DECODER:
+            # encoder only can also benefit from KV cache for prefix caching
```

Comment:
My understanding was that attention in an Encoder is (typically?) bidirectional, so that the encoder had to process the entire sequence at once. If that's the case, it doesn't seem like KV cache makes sense here.

I'm hitting this in my work on encoder-decoder support, since the Attention abstraction in V1 assumes KV cache is used for attention, though it's not relevant for the encoder. I don't have it fully sorted out yet.

If I'm missing why it's needed here, could you help me understand? Thanks!

Reply:
I see. When I had KV cache enabled, I was getting different results out of the encoder between V0 and V1. It's very possible the cause was something other than having KV cache enabled, though.

Just to check though, have you validated that the outputs are consistent between V0 and V1?

Reply:
great! I'm sure my problem was something else then.

FWIW, I did just get encoder-without-kv-cache working in my branch here: https://github.com/vllm-project/vllm/compare/main...russellb:vllm:v1-whisper

The branch is still WIP, but I was able to verify that the output of the encoder matches between V0 and V1 (at least with whisper)

Reply:
Cool,  I'll try it out. I'm especially keen to see if it improves the performance of encoder models to run without kv-cache.